### PR TITLE
fixcpn): use model semver over radio semver for decoding model sources and switches

### LINKED ISSUE #4759: 2.10rc1 conversion issue from 2.9.4
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### What part of EdgeTX is the focus of this bug?

Transmitter firmware

### Current Behavior

I flashed 2.10rc1 on top of 2.9.4. Installed Companion 2.10 on Mac. Read models and settings in Companion. Deleted a few old models, wrote models and settings to TX16s, and in my Avanti Jet configuration I lost RS and LS entries in the mixer, logical switches, and special functions. In the mixer I saw RS (gear) replaced with --  And LS (flaps replaced with -- in multiple locations throughout the configuration.

### Expected Behavior

After flash, specified controls (sliders in my case) remain throughout the configuration.

### Steps To Reproduce

Use the 2.9.4 yaml, on 2.9.4 firmware and verify LS/RS entries in mixer/LS/SF. 
Flash 2.10RC1
Install Companion 2.10
In Companion read models settings. Delete a few models but do not make any changes to Avanti configuration.
Write models/settings to radio.
Note loss of LS/RS in mixer/LS/SF.

### Version

Other (Please specify below)

### Transmitter

RadioMaster TX16S / TX16SMK2

### Operating System (OS)

macOS

### OS Version

2.10RC1 and 2.10 Companion for Mac.

### Anything else?

[Archive.zip](https://github.com/EdgeTX/edgetx/files/14628942/Archive.zip)

### PR DESCRIPTION
**Pull request by elecpower**
Fixes #4759 and also display message for each model file where the version is ahead of Companion.

### CODE REVIEW

File: companion/src/firmwares/edgetx/yaml_modeldata.cpp [hunk 1]
```diff
@@ -1163,8 +1162,13 @@ bool convert<ModelData>::decode(const Node& node, ModelData& rhs)
 
   qDebug() << "Settings version:" << modelSettingsVersion.toString();
 
-  if (modelSettingsVersion > SemanticVersion(VERSION))
-    qDebug() << "Warning: version not supported by Companion!";
+  if (modelSettingsVersion > SemanticVersion(VERSION)) {
+    QString prmpt = QCoreApplication::translate("YamlModelSettings", "Warning: Model settings file version %1 is not supported by this version of Companion!\n\nModel and radio settings may be corrupted if you continue.\n\nI acknowledge and accept the consequences.");
```
Comment (pfeerick) [inline companion/src/firmwares/edgetx/yaml_modeldata.cpp:1166]: Is it possible to display the model filename in question here?

Comment (elecpower) [reply companion/src/firmwares/edgetx/yaml_modeldata.cpp:1166]: A present the best I could do is display the model name and add a TODO reminder.

Comment (pfeerick) [reply companion/src/firmwares/edgetx/yaml_modeldata.cpp:1166]: That would be great in the interim - better than a bunch of non-descript "something ain't right" errors. Just thinking back to the other todo about the error handling, maybe that is where the filename could get exposed as part of the reason for failing.

Comment (elecpower) [reply companion/src/firmwares/edgetx/yaml_modeldata.cpp:1166]: One day.....

**Pull request merged**